src/Promenade.Web/Views/Home/Locations.cshtml

```diff
@@ -0,0 +1,160 @@
+                                                        </td>
+
+                                                    </tr>
+                                                </tbody>
+                                            </table>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </text>
+                    }
+                }
+                else
+                {
+                    <script>
+                        var noResults = true;
+                    </script>
+                }
+            </div>
+        </div>
+        @section scripts {
+            <script>
+        var useMyLocation = 'Use my location';
+
+        $().ready(function () {
+            if (location.protocol != 'https:' && location.hostname != 'localhost') {
+                location.href = 'https:' + window.location.href.substring(window.location.protocol.length);
+            }
+            if (navigator.geolocation) {
+                $('#GeoButton').html(useMyLocation);
+                $('#GeoButton').show();
+            }
+        });
+
+        $('#LocatorAddress').ready(function () {
+            if (typeof noResults !== 'undefined') {
+                $('#LocatorAddress').focus();
+            }
+        });
+
+        var geoSuccess = function (position) {
+            $('#Latitude').val(position.coords.latitude);
+            $('#Longitude').val(position.coords.longitude);
+            $('#GeoForm').submit();
+        }
+
+        var geoFailure = function (error) {
+            $('#GeoButton').html(useMyLocation);
+            $('#GeoButton').prop('disabled', false);
+            alert('There was a problem determining your location, please enter an address or zip code to find your nearest library.');
+            console.log(error.message);
+        }
+
+        $('#GeoButton').on('click', function () {
+            $('#GeoButton').html('<span class="fa fa-spinner fa-spin"></span> ' + useMyLocation);
+            $('#GeoButton').prop('disabled', true);
+            var geoOptions = {
+                enableHighAccuracy: true,
+                maximumAge: 30000,
+                timeout: 27000
+            };
+            navigator.geolocation.getCurrentPosition(geoSuccess, geoFailure, geoOptions);
+        });
+
+        $('#GoButton').on('click', function () {
+            var locatorAddress = $('#LocatorAddress').val();
+            if (!$.trim(locatorAddress).length) {
+                return false;
+            } else {
+                $('#GoButton').html('<span class="fa fa-spinner fa-spin"></span>');
+                $('#GoButton').attr("href",);
+                $('#GoButton').prop('disabled', true);
+                $('#AddressForm').attr('action','/Locations/Find').submit();
```

Comment:
This URL should come from `Url.Action()` as a Razor call which is placed into this JavaScript so that the path isn't hardcoded.